### ISSUE TITLE
Unable to bind to LDAPS domain

### ISSUE THREAD

**Issue by refriedjello** [dalibo/ldap2pg#370]
I just started trying to get ldap2pg to work for me. At the moment I am just trying to connect to the domain with ldap2pg so that queries can be performed. Please note, that the machine is joined to the domain in question using CentrifyDC and that ldapsearch command works as I expect. However when I test my config I receive the following error:

```
$ sudo /usr/local/bin/ldap2pg
Starting ldap2pg 5.5.
Using /etc/ldap2pg.yml.
Failed to connect to LDAP: {'result': -6, 'desc': 'Unknown authentication method', 'errno': 2, 'ctrls': [], 'info': 'SASL(-4): no mechanism available: No worthy mechs found'}
```

Below is the contents of /etc/ldap2pg.yml :

```
postgres:
  dsn: postgres://postgres:password123@127.0.0.1:5432

ldap:
  uri: ldaps://my.domain.com:636
  binddn: "OU=Service Accounts,OU=Users,OU=Internal,DC=my,DC=domain,DC=com"
  user: ldap_service_account
  password: somePASSWORD

sync_map:
  - description: Import all Level99 Users
    ldap:
      base: "CN=Level99,OU=Access Groups,OU=Groups,OU=Internal,DC=my,DC=domain,DC=com"
    role:
      name: '{member.cn}'
      options: LOGIN SUPERUSER
```

I read through the example file, and don't see the "ldap:" section details in that file, so I wasn't sure how best to enter it. Will ldap2pg look at the openldap config files and use those settings if they are defined there? Will all settings defined in the ldap2pg config file overwrite ldap settings? One thing I notice is that I have not defined in ldap2pg my TLS/SSL certificate info.

Thanks!

EDIT:

I am running this on a CentOS 7 server.
The machine is joined to my domain "my.domain.com" and I can login to the server itself using my domain credentials.
I have also setup LDAPS authentication to postgres itself, and can login to the database using psql command, using my domain credentials - I just have to first define a role for my user account manually.

**Comment by refriedjello** [comment]
I was able to get this working. Running the `ldap2pg` command with the `-v` flag showed that it was not reading my openldap client configuration file at /etc/openldap.ldap.conf so I created a symlink for it:

```
sudo mkdir /etc/ldap
sudo ln -s /etc/openldap/ldap.conf /etc/ldap/ldap.conf
```

Then I was able to connect with these settings in ldap2pg.yml for LDAP:

```
ldap:
  uri: ldaps://my.domain.com
  binddn: "CN=ldap_service_account,OU=Service Accounts,OU=Users,OU=Internal,DC=my,DC=domain,DC=com"
  password: somePASSWORD
```

And I used these settings in my `/etc/openldap/ldap.conf` file, where `TLS_CACERT` is a combination PEM file containing both the CA and intermediate CA certs for my domain:

```
TLS_CACERTDIR  /etc/pki/tls/certs
TLS_CACERT      /etc/pki/tls/certs/my_domain_com-combo.crt
TLS_REQCERT     demand
SASL_NOCANON    on
```

Now running the `ldap2pg` command I can that it is able to see users on the domain for import.

**Comment by bersace** [comment]
Hi @refriedjello thanks for reaching.

I'm glad that verbose messages helps you to find the workaround. This issue has been reported at #350. Yesterday, I just released ldap2pg 5.6 which fixes this issue by reading openldap.conf too !

RPM are now available for CentOS 8, 7 and 6 at https://yum.dalibo.org/labs .

Thanks !
